### ISSUE TITLE
0C4 when running mantastic

### ISSUE THREAD

**Issue by gngrossi** [ZOSOpenTools/promptersport#2]
![image](https://github.com/ZOSOpenTools/promptersport/assets/115715798/1e601f5e-324a-4c7a-9d83-1fc45946393a)

![image](https://github.com/ZOSOpenTools/promptersport/assets/115715798/1d8c8b7d-b9d8-4529-87af-1c924c90af9e)

**Comment by gngrossi** [comment]
Running:
gum version 0.13.0
mandb 2.12.0
GNU bash, version 5.2.21(1)-release (i370-ibm-openedition)

Results from:
1) mantastic -h
![image](https://github.com/ZOSOpenTools/promptersport/assets/115715798/b7a02e80-c507-43b6-8566-55f21b101fc7)

2) mantastic diff
![image](https://github.com/ZOSOpenTools/promptersport/assets/115715798/4dc751cf-ac78-4077-a326-dafde9d09d38)

**Comment by gngrossi** [comment]
mantastic -help works as before.

Now seeing an 0C4 with mantastic and mantastic diff
![image](https://github.com/ZOSOpenTools/promptersport/assets/115715798/3363e264-5141-40d1-a668-4a563f854a3a)

**Comment by IgorTodorovskiIBM** [comment]
FYI @lbdyck .

@gngrossi , are you able to add a `set -x` to the 2nd line of `mantastic`.  This will tell us which command is triggering the OC4.

**Comment by gngrossi** [comment]
@IgorTodorovskiIBM
Screen capture from set -x
![image](https://github.com/ZOSOpenTools/promptersport/assets/115715798/72183fe5-12cc-44e4-bb09-4dc4ff5ddf8e)

--

1st page of CEEDUMP without set -x
[mantastic_CEEDUMP.txt](https://github.com/ZOSOpenTools/promptersport/files/14347719/mantastic_CEEDUMP.txt)

Screen capture without set -x
![image](https://github.com/ZOSOpenTools/promptersport/assets/115715798/e30a32a4-9bb3-46ff-8799-775b8002a372)

**Comment by lbdyck** [comment]
This is the version of `bash` that I'm working with and have no issues in case this helps - it's from the z/OS Open Tools install

```
bash --version
GNU bash, version 5.2.21(1)-release (i370-ibm-openedition)
Copyright (C) 2022 Free Software Foundation, Inc.
```

**Comment by IgorTodorovskiIBM** [comment]
Thanks @gngrossi . The OC4 seems to be stemming from `gum`. 

@dustin-ward FYI - Is _rto_s390x_zos a Go function?

**Comment by lbdyck** [comment]
what version of z/OS are you encountering this error on?

I have tested on 2.5 and 3.1 - on my 2.4 system I get a number of CEE.... type errors but haven't followed up as 2.4 won't be around much longer.

**Comment by gngrossi** [comment]
Running z/OS 2.5

10:08:17 RC=(0) [SYSA] bash-5.2$ bash --version
GNU bash, version 5.2.21(1)-release (i370-ibm-openedition)
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

10:08:25 RC=(0) [SYSA] bash-5.2$ mandb --version
mandb 2.12.0

10:08:36 RC=(0) [SYSA] bash-5.2$ gum --version
gum version 0.13.0

**Comment by dustin-ward** [comment]
> @dustin-ward FYI - Is _rto_s390x_zos a Go function?

This is the entry point to the Go runtime. Something is crashing the Go runtime, but I am unable to reproduce the issue

**Comment by gngrossi** [comment]
Upgraded to the current package.

Commands worked first time:
mantastic -h
mantastic diff  (diff nothing appropriate)
mantastic        (received a prompt for search)

Second time:
mantastic -h  (worked)
mantastic diff
![image](https://github.com/ZOSOpenTools/promptersport/assets/115715798/f7e1b6f5-b029-4ebb-860b-ab88bc715797)
mantastic
![image](https://github.com/ZOSOpenTools/promptersport/assets/115715798/f596998d-ec00-4551-b053-fba001ae50a7)

Logged off/on
Same results as the Second time.

**Comment by gngrossi** [comment]
Last part the CEEDUMP.
  errno information :
  Thread Id .... 33D1200000000000 Errno ...... 137 Errnojr .... 05FC0119
CEE3846I CEEDUMP Processing completed.
--

[CEEDUMP_gum.txt](https://github.com/ZOSOpenTools/promptersport/files/14380165/CEEDUMP_gum.txt)

--
References to Errno
https://www.ibm.com/docs/en/zos/2.5.0?topic=codes-return-errnos
137	0089	ENOTTY	The I/O control operator is inappropriate.

https://www.ibm.com/docs/en/zos/2.5.0?topic=errnojrs-zos-unix-reason-codes
05FC	JRICSFnotActive
ICSF must be running to open/read from /dev/random or /dev/urandom.

0119	JRNotSupportedForFileType
The requested service is not supported for this file type.
Action: Reissue the request, specifying a file of the correct type for the request.

**Comment by dustin-ward** [comment]
If we can get an environment dump, the Go on Z team can start taking a look at this internally

**Comment by lbdyck** [comment]
how do i generate an environment dump?

**Comment by lbdyck** [comment]
Unless someone else has this issue I'm not going to pursue it as our 2.4 system is schedule to retire very very soon and the issue does not occur on 2.5 or 3.1

**Comment by dustin-ward** [comment]
> Running z/OS 2.5
> 
> 10:08:17 RC=(0) [SYSA] bash-5.2$ bash --version GNU bash, version 5.2.21(1)-release (i370-ibm-openedition) Copyright (C) 2022 Free Software Foundation, Inc. License GPLv3+: GNU GPL version 3 or later http://gnu.org/licenses/gpl.html This is free software; you are free to change and redistribute it. There is NO WARRANTY, to the extent permitted by law.
> 
> 10:08:25 RC=(0) [SYSA] bash-5.2$ mandb --version mandb 2.12.0
> 
> 10:08:36 RC=(0) [SYSA] bash-5.2$ gum --version gum version 0.13.0

This error is still happening in 2.5 for @gngrossi no?

`printenv` will dump the environment contents to the terminal

**Comment by gngrossi** [comment]
Yes, I'm seeing the 0C4 on a 2.5 system. Soon I will be testing on 3.1.

These commands work:
mantastic -h
mantastic -help
mantastic diff -d

These commands generate an 0C4:
mantastic
mantastic ?
mantastic diff
mantastic git -a push

**Comment by lbdyck** [comment]
This is on a z/OS 2.4 system - no `go` or `gum` involved - just the `zot` version of `man`

```
/home/splbd>man diff
man: command exited with status 137: (cd /home/splbd/zopen/usr/local/share/man && /jenkins/zopen/usr/local/zopen/man/man-db-2.12.0/libexec/man-db/zsoelim) | (cd /home/splbd/zopen/usr/local/share/man && /jenkins/zopen/usr/local/zopen/man/man-db-2.12.0/libexec/man-db/manconv -f UTF-8:ISO-8859-1 -t UTF-8//IGNORE) | (cd /home/splbd/zopen/usr/local/share/man && preconv -e UTF-8) | (cd /home/splbd/zopen/usr/local/share/man && tbl) | (cd /home/splbd/zopen/usr/local/share/man && nroff -mandoc -rLL=138n -rLT=138n [-Tutf8)
```

[manenv.txt](https://github.com/ZOSOpenTools/promptersport/files/14438135/manenv.txt)

**Comment by gngrossi** [comment]
Here's the env output.
[env_output.txt](https://github.com/ZOSOpenTools/promptersport/files/14438735/env_output.txt)

**Comment by gngrossi** [comment]
After two updates to my environment, I ran through these use cases without an 0C4.
1) unset STEPLIB   (export STEPLIB=SYS1.SCCNCMP)
2) sourced a profile /hewitt/zopentools/oelcpp20/oelcpp.profile   for C/C++ for Open Enterprise Languages (pax file installed)

**Use cases:** with expected results
mantastic -h
mantastic -help
mantastic diff -d  (result was the gum pager for diff)

**Use cases without abends**
mantastic 
Enter a command for a man page or a keyword to search for: diff
diff: nothing appropriate.
No manual entry for no
provide some content to display

mantastic diff
diff: nothing appropriate.
No manual entry for no
provide some content to display

**Seeing a set of man pages for CoZ ? After a selection, the result was the gum pager for it.**
mantastic ?
Select Man Page to View
> catsearch
  cozclient
  cozcontrol
  cozsftp
  fromdsn
  fromfile
  safauth
  todsn
  tofile
  zsym

mantastic git -a push
push: nothing appropriate.
No manual entry for no
provide some content to display
